lib/facter/custom_facts/util/fact.rb

```diff
@@ -201,11 +203,27 @@ def find_first_real_value(resolutions)
             break
           end
           @used_resolution_weight = resolve.weight
-          return value unless value.nil?
+          next if value.nil?
+
+          log_fact_path(resolve)
+
+          return value
         end
         nil
       end
 
+      def log_fact_path(resolve)
+        fact_type, resolved_from = if resolve.instance_of?(Facter::Core::Aggregate)
+                                     ['aggregate', resolve.instance_variable_get(:@aggregate).source_location[0]]
```

Comment:
This is causing CI failures as `source_location`  can be nil. It would be good to add tests for this change also.